lib/request.js

```diff
@@ -39,6 +41,17 @@ Object.defineProperties(Request.prototype, {
     get: function () {
       return this.raw.id
     }
+  },
+  protocol: {
+    get: function () {
+      // https://nodejs.org/api/tls.html#tls_tlssocket_encrypted
+      const protocol = this.raw.socket.encrypted ? 'https' : 'http'
```

Comment:
Note that we could have trouble in tests using `inject`: https://github.com/fastify/light-my-request/issues/87